### ISSUE TITLE
upnpc-static.exe reported as trojan

### ISSUE THREAD

**Issue by HelloZeroNet** [HelloZeroNet/ZeroNet#5]
```
Norton

File blocked: tools\upnpc\upnpc-static.exe 

Threat name: Trojan.Gen.2 ( http://www.symantec.com/security_response/writeup.jsp?docid=2011-082216-3542-99 ) 
```

Pure python upnp implementation would be nice, but i could not find any good.

**Comment by shakna-israel** [comment]
[PyUPnP](https://github.com/fuzeman/PyUPnP) might be a place to start, though it adds [Twisted](https://twistedmatrix.com/trac/) as a dependency. (Of course, Twisted itself should be able to cover this, but who wants to write from scratch?)

**Comment by sirMackk** [comment]
Hi, I can look into this. It seems related to #20 as well, so might as well kill two birds with one stone. Do you know if someone is working on this or is it safe to dive in?

**Comment by HelloZeroNet** [comment]
I think nobody is working on it, so its free to go :)
I want avoid add extra dependencies, so it would be really nice to make it work using only gevent+standard libs.

**Comment by sirMackk** [comment]
Sounds great, thanks for the pointer!

**Comment by sirMackk** [comment]
Hi! I *think* I got it. It uses gevent + the stdlib. I've tested it with two UPnP enabled routers I have access to and it worked both times. I'll be able to test it with a third one this weekend. 

Before I do final cleanup and submit a PR, do you know someone who could test my branch against their router/s? I don't have a lot of experience with UPnP and I'm trying to cover anything I might have missed. My branch: https://github.com/sirMackk/ZeroNet/tree/upnp_punch .

**Comment by HelloZeroNet** [comment]
Wow looks nice, good work! I'm going to test it later today.

**Comment by HelloZeroNet** [comment]
Just tried it and its working here! :) On Monday i'm going to test it on other routers.

**Comment by sirMackk** [comment]
Oh man, that's great! I posted a message on ZeroNet to see if anyone else wants to volunteer to test this functionality out. Might make a ZeroNet site to coordinate this type of effort.

**Comment by HelloZeroNet** [comment]
it's working at my workplace too :) So I think we can include it in the next release if its ok for you too.

**Comment by sirMackk** [comment]
Great! Ive fixed some small platform related bugs as well. I'll submit a PR
as soon as I get back from work. Thanks!

**Comment by HelloZeroNet** [comment]
Fixed in 34f6d1ee7c8b5a8958f675c5e17a95c34c6a4967
